### PR TITLE
[13.0][ADD] hr_recruitment_notification: subscribe users to new applications automatically

### PR DESCRIPTION
**Pull request by joao-p-marques**
New module that provides a new message subtype to allow users to be automatically subscribed to new applications from a Job Position.

@Tecnativa
TT30413

ping @pedrobaeza @Yajo

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

File: hr_recruitment_notification/data/message_subtypes.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2021 Tecnativa - João Marques -->
+<odoo>
+    <data noupdate="1">
```
Comment (yajo) [inline hr_recruitment_notification/data/message_subtypes.xml:4]: It seems to me like this shouldn't be noupdate.

File: hr_recruitment_notification/data/message_subtypes.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        <!-- Message Subtype related to Job Positions -->
+        <record id="mt_hr_applicant_new" model="mail.message.subtype">
+            <field name="name">New Applicant</field>
+            <field name="description">Applicant created</field>
+            <field name="res_model">hr.job</field>
+            <field name="default" eval="False" />
+            <field name="parent_id" eval="ref('hr_recruitment.mt_applicant_new')" />
```
Comment (yajo) [inline hr_recruitment_notification/data/message_subtypes.xml]:
```suggestion
            <field name="parent_id" ref="hr_recruitment.mt_applicant_new" />
```

**Review by yajo — CHANGES_REQUESTED** [review]
Please open PR for upstream odoo master about this.

File: hr_recruitment_notification/data/message_subtypes.xml (hunk 1 above)
Comment (joao-p-marques) [reply hr_recruitment_notification/data/message_subtypes.xml:4]: Well, fine by me, it just seems that that is the standard in all Odoo message subtypes. For example, in a runbot with all core modules installed:

image

**Comment by joao-p-marques** [comment]
Thanks @Yajo!

Changes made and PR opened to Odoo in https://github.com/odoo/odoo/pull/75736

File: hr_recruitment_notification/data/message_subtypes.xml (hunk 1 above)
Comment (yajo) [reply hr_recruitment_notification/data/message_subtypes.xml:4]: Still missing.

**Review by yajo — CHANGES_REQUESTED** [review]

File: hr_recruitment_notification/data/message_subtypes.xml (hunk 1 above)
Comment (pedrobaeza) [reply hr_recruitment_notification/data/message_subtypes.xml:4]: They are noupdate in core, and they should be that way for allowing any modification (for example, `default` flag).

Comment (yajo) [reply hr_recruitment_notification/data/message_subtypes.xml:4]: Ah that makes sense.

**Review by yajo — APPROVED** [review]

**Pull request merged**